File: php/features.php
<?php
declare(strict_types=1);

// CreditCardValidation SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CreditCardValidationFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CreditCardValidationBaseFeature();
            case "test":
                return new CreditCardValidationTestFeature();
            default:
                return new CreditCardValidationBaseFeature();
        }
    }
}

File: py/features.py
# CreditCardValidation SDK feature factory

from feature.base_feature import CreditCardValidationBaseFeature
from feature.test_feature import CreditCardValidationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CreditCardValidationBaseFeature(),
        "test": lambda: CreditCardValidationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
